nimare/meta/ibma.py

```diff
@@ -1542,3 +1535,152 @@ def correct_fwe_montecarlo(self, result, n_iters=5000, n_cores=1):
+        )
+        return description
+
+    def _fit_model(self, t_maps, study_mask=None):
+        """Fit the model to the data."""
+        n_studies, n_voxels = t_maps.shape
+
+        if study_mask is None:
+            # If no mask is provided, assume all studies are included. This is always the case
+            # when using the aggressive mask.
+            study_mask = np.arange(n_studies)
+
+        sample_sizes = np.array([np.mean(self.inputs_["sample_sizes"][idx]) for idx in study_mask])
+        n_maps = np.tile(sample_sizes, (n_voxels, 1)).T
+
+        # Calculate Hedge's g maps: Standardized mean
+        cohens_maps = t_to_d(t_maps, n_maps)
+        hedges_maps, var_hedges_maps = d_to_g(cohens_maps, n_maps, return_variance=True)
+
+        del n_maps, sample_sizes, cohens_maps
+
+        pymare_dset = pymare.Dataset(y=hedges_maps, v=var_hedges_maps)
+        est = pymare.estimators.WeightedLeastSquares(tau2=self.tau2)
+        est.fit_dataset(pymare_dset)
+        est_summary = est.summary()
+
+        fe_stats = est_summary.get_fe_stats()
+        z_map = fe_stats["z"].squeeze()
+        p_map = fe_stats["p"].squeeze()
+        est_map = fe_stats["est"].squeeze()
+        se_map = fe_stats["se"].squeeze()
+        dof_map = np.tile(n_studies - 1, n_voxels).astype(np.int32)
+
+        return z_map, p_map, est_map, se_map, dof_map
+
+    def _fit(self, dataset):
+        self.dataset = dataset
+        self.masker = self.masker or dataset.masker
+        if not isinstance(self.masker, NiftiMasker):
+            LGR.warning(
+                f"A {type(self.masker)} mask has been detected. "
+                "Masks which average across voxels will likely produce biased results when used "
+                "with this Estimator."
+            )
+
+        if self.aggressive_mask:
+            voxel_mask = self.inputs_["aggressive_mask"]
+            result_maps = self._fit_model(self.inputs_["t_maps"][:, voxel_mask])
+
+            z_map, p_map, est_map, se_map, dof_map = tuple(
+                map(lambda x: _boolean_unmask(x, voxel_mask), result_maps)
+            )
+        else:
+            n_voxels = self.inputs_["t_maps"].shape[1]
+
+            z_map, p_map, est_map, se_map = [np.zeros(n_voxels, dtype=float) for _ in range(4)]
+            dof_map = np.zeros(n_voxels, dtype=np.int32)
+
+            for bag in self.inputs_["data_bags"]["t_maps"]:
+                (
+                    z_map[bag["voxel_mask"]],
+                    p_map[bag["voxel_mask"]],
+                    est_map[bag["voxel_mask"]],
+                    se_map[bag["voxel_mask"]],
+                    dof_map[bag["voxel_mask"]],
+                ) = self._fit_model(bag["values"], bag["study_mask"])
+
+        # tau2 is a float, not a map, so it can't go into the results dictionary
+        tables = {"level-estimator": pd.DataFrame(columns=["tau2"], data=[self.tau2])}
```

Comment:
this is fine for now since it's how weightedleastsquares returns results, but it is awkward to return a single value in a pandas dataframe versus it just being an attribute of the object.